File: backend_pruebas/python/fastappi/app/models/task.py
# app/models/task.py
from sqlalchemy import Column, Integer, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database.db import Base

class Task(Base):
    __tablename__ = 'tasks'
    
    id = Column(Integer, primary_key=True, index=True)
    parent_id = Column(Integer, ForeignKey('tasks.id'), nullable=True)
    progress = Column(Float, default=0.0)
    estimated_time = Column(Integer)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    parent = relationship("Task", remote_side=[id], backref="subtasks")

File: backend_pruebas/go/puro_go/models/task.go
package models

import "time"

type Task struct {
	ID            int        `json:"id"`
	ParentID      *int       `json:"parent_id,omitempty"` // Puede ser nulo
	Progress      float64    `json:"progress"`
	EstimatedTime int        `json:"estimated_time"` // en segundos
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}
